SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp

```diff
@@ -1,7 +1,399 @@
 
+#include "SpatialViewUtils.h"
 #include "Tests/TestDefinitions.h"
 
+#include "SpatialView/OpList/EntityComponentOpList.h"
 #include "SpatialView/ViewDelta.h"
+#include "Tests/SpatialView/ExptectedViewDelta.h"
 
 #define VIEWDELTA_TEST(TestName) GDK_TEST(Core, ViewDelta, TestName)
+
+using namespace SpatialGDK;
+
+namespace
+{
+void SetFromOpList(ViewDelta& Delta, EntityView& View, EntityComponentOpListBuilder& OpListBuilder)
+{
+	OpList Ops = MoveTemp(OpListBuilder).CreateOpList();
+	TArray<OpList> OpLists;
+	OpLists.Push(MoveTemp(Ops));
+	Delta.SetFromOpList(MoveTemp(OpLists), View);
+}
+} // anonymous namespace
+
+VIEWDELTA_TEST(GIVEN_empty_view_WHEN_add_entity_THEN_get_entity_in_view_and_delta)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+
+	EntityComponentOpListBuilder OpListBuilder;
+	OpListBuilder.AddEntity(TestEntityId);
+	SetFromOpList(InputDelta, InputView, OpListBuilder);
+
+	EntityView ExpectedView;
+	AddEntityToView(ExpectedView, TestEntityId);
+
+	ExpectedViewDelta ExpectedDelta;
+	ExpectedDelta.AddEntityDelta(TestEntityId, ExpectedViewDelta::ADD);
+
+	TestTrue("View Deltas are equal", ExpectedDelta.Compare(InputDelta));
+	TestTrue("Views are equal", AreEquivalent(InputView, ExpectedView));
+
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_entity_in_view_WHEN_remove_entity_THEN_empty_view)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+	AddEntityToView(InputView, TestEntityId);
+
+	EntityComponentOpListBuilder OpListBuilder;
+	OpListBuilder.RemoveEntity(TestEntityId);
+	SetFromOpList(InputDelta, InputView, OpListBuilder);
+
+	EntityView ExpectedView;
+	ExpectedViewDelta ExpectedDelta;
+	ExpectedDelta.AddEntityDelta(TestEntityId, ExpectedViewDelta::REMOVE);
+
+	TestTrue("View Deltas are equal", ExpectedDelta.Compare(InputDelta));
+	TestTrue("Views are equal", AreEquivalent(InputView, ExpectedView));
+
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_entity_in_view_WHEN_add_component_THEN_entity_and_component_in_view)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+	AddEntityToView(InputView, TestEntityId);
+
+	EntityComponentOpListBuilder OpListBuilder;
+	OpListBuilder.AddComponent(TestEntityId, CreateTestComponentData(TestComponentId, TestComponentValue));
+	SetFromOpList(InputDelta, InputView, OpListBuilder);
+
+	EntityView ExpectedView;
+	AddEntityToView(ExpectedView, TestEntityId);
+	AddComponentToView(ExpectedView, TestEntityId, TestComponentId, TestComponentValue);
+
+	ExpectedViewDelta ExpectedDelta;
+	ExpectedDelta.AddEntityDelta(TestEntityId, ExpectedViewDelta::UPDATE)
+		.AddComponentAdded(TestEntityId, CreateTestComponentData(TestComponentId, TestComponentValue));
+
+	TestTrue("View Deltas are equal", ExpectedDelta.Compare(InputDelta));
+	TestTrue("Views are equal", AreEquivalent(InputView, ExpectedView));
+
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_entity_and_component_in_view_WHEN_update_component_THEN_component_udated_in_view)
+{
+	double NewComponentValue = 30;
```

Comment:
nit: if the other value fixtures are going to be outside of the tests, would rather this was consistent - `OtherTestComponentValue` or something (also elsewhere in the file)